.github/workflows/close-issues.yml

```diff
@@ -0,0 +1,41 @@
+name: Close issues
+
+on:
+  schedule:
+    - cron: '0 1 * * *'
+  workflow_dispatch:
+
+permissions:
+  contents: read
+
+# Ensure scripts are run with pipefail. See:
+# https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#exit-codes-and-error-action-preference
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  close-issues:
+    runs-on: ubuntu-latest
+    if: github.repository == 'microsoft/TypeScript'
+    permissions:
+      contents: read # Apparently required to create issues
+      issues: write
+
+    steps:
+      - name: Close issues
+        run: |
+          DATE=$(date --date='2 days ago' --iso-8601)
+
+          close_issues() {
+            echo "Closing issues marked as '$1'."
+            for issue in $(gh issue list --label "$1" --repo ${{ github.repository }} --state open --search "updated:<$DATE" --json number --jq '.[].number'); do
+              echo "Closing https://github.com/${{ github.repository }}/issues/$issue"
+              gh issue close $issue --reason "not planned" --comment "This issue has been marked as '$1' and has seen no recent activity. It has been automatically closed for house-keeping purposes."
+            done
+          }
+
+          close_issues "Duplicate"

```

Comment:
```suggestion
          close_issues "Duplicate"
          close_issues "Unactionable"
          close_issues "Not a Defect"
```

Reply:
I had been meaning to ask about those; we don't auto-close them but feels like we should. I'll tack it on.

Reply:
There are some 80 issues labeled with "Not a Defect" that are still open; hopefully it won't get rate limited.